Modules/HMPID/src/HmpidTaskMatches.cxx

```diff
@@ -0,0 +1,185 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// // distributed under the terms of the GNU General Public License v3 (GPL
+// // Version 3), copied verbatim in the file "COPYING".
+// //
+// // See http://alice-o2.web.cern.ch/license for full licensing information.
+// //
+// // In applying this license CERN does not waive the privileges and immunities
+// // granted to it by virtue of its status as an Intergovernmental Organization
+// // or submit itself to any jurisdiction.
+//
+
+///
+/// \file   HmpidTaskMatches.cxx
+/// \author Nicola Nicassio, Giacomo Volpe
+///
+
+#include <TCanvas.h>
+#include <TH1.h>
+#include <TH2.h>
+
+#include "QualityControl/QcInfoLogger.h"
+#include "HMPID/HmpidTaskMatches.h"
+#include <Framework/InputRecord.h>
+#include <Framework/InputRecordWalker.h>
+#include <Framework/InputSpec.h>
+#include "GlobalTrackingWorkflowHelpers/InputHelper.h"
+#include "ReconstructionDataFormats/TrackTPCITS.h"
+#include "DataFormatsTPC/TrackTPC.h"
+#include "DataFormatsGlobalTracking/RecoContainerCreateTracksVariadic.h"
+#include "ReconstructionDataFormats/TrackParametrization.h"
+#include "DetectorsBase/Propagator.h"
+
+#include "DataFormatsHMP/Cluster.h"
+#include "ReconstructionDataFormats/MatchInfoHMP.h"
+
+#ifdef WITH_OPENMP
+#include <omp.h>
+#endif
+
+namespace o2::quality_control_modules::hmpid
+{
+
+HmpidTaskMatches::HmpidTaskMatches() : TaskInterface() {}
+
+HmpidTaskMatches::~HmpidTaskMatches()
+{
+  ILOG(Info, Support) << "destructor called" << ENDM;
+  for (int iCh = 0; iCh < 7; iCh++) {
+    delete mMatchInfoResidualsXTrackMIP[iCh];
+    delete mMatchInfoResidualsYTrackMIP[iCh];
+    delete mMatchInfoChargeClusterMIP[iCh];
+    delete mMatchInfoChargeClusterPhotons[iCh];
+    delete mMatchInfoClusterMIPMap[iCh];
+    delete mMatchInfoThetaCherenkovVsMom[iCh];
+  }
+}
+
+void HmpidTaskMatches::initialize(o2::framework::InitContext& /*ctx*/)
+{
+  ILOG(Debug, Devel) << "initialize TaskMatches" << ENDM;
+
+  mDataRequest = std::make_shared<o2::globaltracking::DataRequest>();
+  mDataRequest->requestTracks(mSrc, false);
+
+  // getJsonParameters();
+
+  BookHistograms();
+
+  // publish histograms (Q: why publishing in initalize?)
+  for (int iCh = 0; iCh < 7; iCh++) {
+
+    getObjectsManager()->startPublishing(mMatchInfoResidualsXTrackMIP[iCh]);
+    getObjectsManager()->startPublishing(mMatchInfoResidualsYTrackMIP[iCh]);
+    getObjectsManager()->startPublishing(mMatchInfoChargeClusterMIP[iCh]);
+    getObjectsManager()->startPublishing(mMatchInfoChargeClusterPhotons[iCh]);
+    getObjectsManager()->startPublishing(mMatchInfoClusterMIPMap[iCh]);
+    getObjectsManager()->setDefaultDrawOptions(mMatchInfoClusterMIPMap[iCh], "colz");
+    getObjectsManager()->setDisplayHint(mMatchInfoClusterMIPMap[iCh], "colz");
+    getObjectsManager()->startPublishing(mMatchInfoThetaCherenkovVsMom[iCh]);
+  }
+
+  ILOG(Info, Support) << "START DOING QC HMPID Matches" << ENDM;
```

Comment:
please remove this log or make it Debug/Trace.

Reply:
Removed!